great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1491,7 +1473,14 @@ def _initialize_usage_statistics(
 
     def _init_datasources(self) -> None:
         """Initialize the datasources in store"""
-        for datasource_name in self._datasource_store.list_keys():
+        datasource_keys: List[
+            Union[str, Tuple[str, str]]
+        ] = self._datasource_store.list_keys()
+
+        if datasource_keys and isinstance(datasource_keys[0], tuple):
+            datasource_keys = [key[1] for key in datasource_keys]
```

Comment:
Don't love this - the `DatasourceStore` requires a `DataContextVariableKey` which is a tuple of variable type and variable name (`(datasources, my_datasource)`).

We just want to be able to extract the name - perhaps there's a way to do this without this extra conditional?

Reply:
Maybe add accessors for `resource_name` and `resource_type` to `DataContextVariableKey` would help make it more clear?

Reply:
so at this point in time, the key has been converted into a tuple and loses all strong typing.

Reply:
There is a Typed NameTuple.
https://docs.python.org/3/library/typing.html#typing.NamedTuple